pkg/reconcilermanager/controllers/jsonpath_test.go

```diff
@@ -0,0 +1,426 @@
+        - "--feature-gates=-exporter.googlecloud.OTLPDirect"
+        resources:
+          requests:
+            cpu: 10m
+            ephemeral-storage: 1Gi
+            memory: 100Mi
+        securityContext:
+          allowPrivilegeEscalation: false
+          readOnlyRootFilesystem: true
+          capabilities:
+            drop:
+            - NET_RAW
+        ports:
+        - containerPort: 55678 # Default OpenCensus receiver port.
+          protocol: TCP
+        - containerPort: 8888  # Metrics.
+          protocol: TCP
+        volumeMounts:
+        - name: otel-agent-config-vol
+          mountPath: /conf
+        livenessProbe:
+          httpGet:
+            path: /
+            port: 13133 # Health Check extension default port.
+            scheme: HTTP
+          timeoutSeconds: 1 # different from declared 
+          periodSeconds: 10 # different from declared
+          successThreshold: 1 # different from declared
+          failureThreshold: 3 # different from declared
+        readinessProbe:
+          httpGet:
+            path: /
+            port: 13133 # Health Check extension default port.
+            scheme: HTTP
+          timeoutSeconds: 1 # different from declared
+          periodSeconds: 10 # different from declared
+          successThreshold: 1 # different from declared
+          failureThreshold: 3 # different from declared
+        terminationMessagePath: "/dev/termination" # different from declared
+        terminationMessagePolicy: FallbackToLogsOnError # different from declared
+        imagePullPolicy: Always # different from declared
+      restartPolicy: Never # different from declared
+      terminationGracePeriodSeconds: 30 # different from declared
+      dnsPolicy: ClusterFirst # different from declared
+      schedulerName: default-scheduler # different from declared
+      volumes:
+      - name: repo
+        emptyDir: {}
+      - name: kube
+        emptyDir: {}
+      securityContext:
+        fsGroup: 65533
+        runAsUser: 1000
+        runAsNonRoot: true
+        seccompProfile:
+          type: RuntimeDefault
+  revisionHistoryLimit: 15 # different from declared
+  progressDeadlineSeconds: 300 # different from declared
+  paused: true # different from declared
+`
+
+func TestIsDeploymentSame(t *testing.T) {
+	declared := yamlToDeployment(t, declaredDeployment)
+	unObjCurrent := yamlToUnstructured(t, currentDeployment)
+	testCases := map[string]struct {
+		declared          *appsv1.Deployment
+		unObjCurrent      *unstructured.Unstructured
+		isAutopilot       bool
+		overrideLimit     bool
+		manualChangeLimit bool
+		isSame            bool
+	}{
+		"non-Autopilot, no override limits, no manual change limits": {
+			declared:          declared,
+			unObjCurrent:      unObjCurrent,
+			isAutopilot:       false,
+			overrideLimit:     false,
+			manualChangeLimit: false,
+			isSame:            true,
+		},
+		"non-Autopilot, no override limits, manual change limits": {
```

Comment:
For posterity, add a comment to explain why we allow manual limit change when limit is not specified.

Reply:
Done.